grpc/src/main/resources/assets/pt/src/view/App.js

```diff
@@ -130,13 +131,15 @@ export default class App extends React.Component {
                                 console.log(response.toObject());
                                 const paths = response.getPathsList().map(path => new Path(path));
                                 const selectedPath = component._selectPathOnReceive(paths);
+                                const queryTimeSeconds = (Date.now() - startTime) * 1.0 / 1000;
```

Comment:
nit: react executes `setState` updater functions asynchronously for performance reasons, so by grabbing the end time within the updater fcn, we could mistakenly include some UI-updating-time in the reported query time. for a little more accuracy, we could calculate the request time within the `router.routePt` callback but outside the `setState` updater function

Reply:
Good note, totally makes sense! Just moved this outside of the `setState` callback, and tested locally that things look good